---
author: Elizabeth Czarniak (CZARNIA_ELIZ@bentley.edu)
---

We're going to use fake data here for illustrative purposes. You can replace our fake data with your real data in the code below. 

We'll put the data into a dataframe and then make a variable with a list of the independent variables and a variable with the outcome variable.

In [ ]:
import pandas as pd

# Replace this fake data with your real data
df = pd.DataFrame( {
    'x1':[2, 7, 4, 3, 11, 18, 6, 15, 9, 12],
    'x2':[4, 6, 10, 1, 18, 11, 8, 20, 16, 13],
    'x3':[11, 16, 20, 6, 14, 8, 5, 23, 13, 10],
    'y':[24, 60, 32, 29, 90, 45, 130, 76, 100, 120]
} )

xs = df[['x1', 'x2', 'x3']]  # list of independent variables
y = df['y']                  # dependent variable

We can use StatsModels' `OLS` to build our multiple linear regression model. We'll print out the coefficients and the intercept, and the coefficients will be in the form of an array when we print them.

In [ ]:
import statsmodels.api as sm

# Add a constant to the dependent variables first
xs = sm.add_constant(xs)

# Build the model
model = sm.OLS(y, xs).fit()

# Show the model summary to get the coefficients and the intercept
model.summary()

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.594
Model:                            OLS   Adj. R-squared:                  0.390
Method:                 Least Squares   F-statistic:                     2.921
Date:                Sat, 07 Aug 2021   Prob (F-statistic):              0.122
Time:                        02:43:14   Log-Likelihood:                -45.689
No. Observations:                  10   AIC:                             99.38
Df Residuals:                       6   BIC:                             100.6
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         77.2443     27.366      2.823      0.0

The coefficients and intercept appear on the left hand side of the output, about half way down, under the heading "coef."

Thus the multiple linear regression model from the example data is $\hat y = 77.2443 - 2.7009x_1  +  7.2989x_2  - 4.8607x_3$.